src/main/kotlin/no/nav/tilleggsstonader/sak/vedtak/OpphørValidatorService.kt

```diff
@@ -0,0 +1,81 @@
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.brukerfeilHvis
+import no.nav.tilleggsstonader.sak.vedtak.barnetilsyn.beregning.TilsynBarnBeregningService
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.VilkårService
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vilkår
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.VilkårStatus
+import no.nav.tilleggsstonader.sak.vilkår.stønadsvilkår.domain.Vilkårsresultat
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.VilkårperiodeService
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.ResultatVilkårperiode
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.domain.Vilkårperioder
+import no.nav.tilleggsstonader.sak.vilkår.vilkårperiode.felles.Vilkårstatus
+import org.springframework.stereotype.Service
+import java.time.LocalDate
+
+@Service
+class OpphørValidatorService(
+    private val vilkårsperiodeService: VilkårperiodeService,
+    private val vilkårService: VilkårService,
+    private val tilsynBarnBeregningService: TilsynBarnBeregningService,
+) {
+
+    fun validerOpphør(saksbehandling: Saksbehandling) {
+        val vilkår = vilkårService.hentVilkår(saksbehandling.id)
+        val vilkårperioder = vilkårsperiodeService.hentVilkårperioder(saksbehandling.id)
+
+        validerIngenNyeOppfylteVilkårEllerVilkårperioder(vilkår, vilkårperioder)
+        validerIngenUtbetalingEtterOpphør(saksbehandling)
+        validerIngenEndredePerioderMedTomEtterOpphørsdato(
+            vilkårperioder,
+            vilkår,
+            saksbehandling.revurderFra ?: error("RevurderFra er påkrevd for opphør"),
+        )
+    }
+
+    private fun validerIngenNyeOppfylteVilkårEllerVilkårperioder(
+        vilkår: List<Vilkår>,
+        vilkårperioder: Vilkårperioder,
+    ) {
+        val finnesOppfyltVilkårEllerVilkårperioderMedStatusNy =
+            vilkår.any { it.status == VilkårStatus.NY && it.resultat == Vilkårsresultat.OPPFYLT } ||
+                vilkårperioder.målgrupper.any { it.status == Vilkårstatus.NY && it.resultat == ResultatVilkårperiode.OPPFYLT } ||
+                vilkårperioder.aktiviteter.any { it.status == Vilkårstatus.NY && it.resultat == ResultatVilkårperiode.OPPFYLT }
+
+        brukerfeilHvis(finnesOppfyltVilkårEllerVilkårperioderMedStatusNy) { "Det er vilkår eller vilkårperiode med vilkårstatus NY og resultat OPPFYLT." }
+    }
+
+    private fun validerIngenUtbetalingEtterOpphør(saksbehandling: Saksbehandling) {
+        val beregningsresultatTilsynBarn = tilsynBarnBeregningService.beregn(saksbehandling)
+        val revurderFraDato = saksbehandling.revurderFra
+        beregningsresultatTilsynBarn.perioder.forEach { periode ->
+            periode.beløpsperioder.forEach {
+                brukerfeilHvis(it.dato > revurderFraDato && it.beløp > 0) { "Det er utbetalinger etter opphørsdato" }
+            }
+        }
+    }
+
+    private fun validerIngenEndredePerioderMedTomEtterOpphørsdato(
```

Comment:
Det er fortsatt gyldig med vilkårperioder eller vilkår som gjelder etter opphørsdatoet.
Eks så kan årsaken til opphør være at man ikke lengre har målgruppe eller en aktivitet, mens man fortsatt har den andre. Dvs målgruppen er opphørt, men man har fortsatt aktiviteten.

Eller man har fortsatt målgruppe eller aktivitet, men ikke utgifter

Det man kan validere her, er at "tom" på en periode ikke er forlenget fra forrige behandlingen. Dvs for vilkår X, så er ikke `X.tom > (X på forrige behandling).tom`